### ISSUE TITLE
CameraCapturer keeps reference to previewContainer even after stopPreview is called

### ISSUE THREAD

**Issue by mwajeeh** [twilio/video-quickstart-android#14]
I don't know where to report this issue with twilio sdk. Point me in right direction if there is any forum where I can report issues with sdk.
We are supporting PIP overlay using twilio sdk, instead of just using sdk in activity and tying it to activity lifecycle. But (by using https://github.com/square/leakcanary) I noticed that `CameraCapturer` keeps reference to `previewContainer` even after `stopPreview` is called causing a memory leak when activity gets destroyed. We keep `Conversation` and `CameraCapturer` objects in service and attach/detach views giving us the flexibility to keep call running even after activity gets destroyed.

I am also facing some other issues and I don't know where to report them.

Here is the example of PIP overlay in our app:

![screenshot_2016-06-21-23-08-28](https://cloud.githubusercontent.com/assets/6572067/16241211/22d36d58-3806-11e6-8245-bd4aa32b6e26.png)

![screenshot_2016-06-21-23-08-48](https://cloud.githubusercontent.com/assets/6572067/16241172/e6a07826-3805-11e6-8808-9d796289f094.png)

**Comment by aaalaniz** [comment]
Hi @mwajeeh,

I'll be happy to answer any questions you have. For future reference, you can email help@twilio.com. 

Have you tried calling `removeAllViews` on the ViewGroup passed into `startPreview` This could be a work around for the issue you are seeing. Also, if you are having trouble with PIP you might try using our latest addition to the `VideoViewRenderer` `applyZOrder(boolean overlaySurface)`. This helps with layering your VideoViewRenderers to support PIP.

You can find the documentation at...
https://media.twiliocdn.com/sdk/android/conversations/releases/0.12.1/docs/

**Comment by mwajeeh** [comment]
Thanks for replying:
> Have you tried calling removeAllViews on the ViewGroup passed into startPreview This could be a work around for the issue you are seeing.

TLDR; `CameraCapturer` doesn't set `previewContainer` to `null` in `stopPreview`, which it should because its no longer needed. I can't do anything to destroy that reference without setting conversation to `null`. 

Let me explain again as these kind of memory leaks would limit usage of this library to only be used in `Activity` and not in `Service`.

Our activity starts a service which has 'Conversation' object. Service starts video call and activity binds ui to conversation to show video. But when activity goes in background we unregister ui using:

```
localVideoTrack.removeRenderer(localVideoRenderer);
localVideoRenderer.release();
videoTrack.removeRenderer(participantVideoRenderer);
 participantVideoRenderer.release(); 
```
And when activity resumes again we add renderers again, which works fine.
Problem is the leak that happens in this process with `CameraCapturer` object inside `Conversation`. 

Solution is simple, just set `previewContainer` to `null` in `CameraCapturer` whenever `stopPreview` is called.

**Comment by aaalaniz** [comment]
@mwajeeh ...Thanks for the detailed explanation. I have corrected this and it will be shipped in the next release.

**Comment by mwajeeh** [comment]
Thanks @aaalaniz , I would highly suggest to change `video-quickstart-android ` such that all code is moved into a `Service` and allow user to come back to ongoing conversation by tapping on ongoing call notification. This is really easy vs showing PIP and it will truly test this awesome sdk.

**Comment by aaalaniz** [comment]
This is great feedback @mwajeeh. Our internal test application uses the SDK a little more extensively. This application is meant to just get users off the ground. You can expect in the future we will add separate applications to show how to use the SDK in other ways.

**Comment by aaalaniz** [comment]
Closing should be resolved with 0.12.2 release
